include/random_walks/gaussian_hamiltonian_monte_carlo_exact_walk.hpp

```diff
@@ -89,6 +89,9 @@ struct Walk
         unsigned int n = P.dimension();
         NT T;
 
+        //normalize the Polyope
+        P.normalize();
```

Comment:
The walk should not (ideally) change the Polytope! 
This is a design mistake since `P` should be a `const &`. 
However, changing $P$ should be avoided.